app/assets/javascripts/discourse/app/lib/autocomplete.js

```diff
@@ -114,6 +115,10 @@ export default function (options) {
   const isInput = me[0].tagName === "INPUT" && !options.treatAsTextarea;
   let inputSelectedItems = [];
 
+  options.textManipulation ??= new TextareaTextManipulation(getOwner(me[0]), {
```

Comment:
@davidtaylorhq I have a doubt this getOwner here is not going to cause issues. Maybe we should just pass some owner to autocomplete where we instanciate it?

Reply:
Hard to say... What does `me[0]` actually refer to? and how is its owner set in the first place? 🤔

Reply:
I think me[0] is a DOM node here @renato ?

Reply:
Then I don't see how it would have an Ember owner associated with it... so `getOwner()` would just return null?

Reply:
Yes that's why that sounded surprising to me. But clearly you can see on the line above we do `me[0].tagName` which is a DOM node API. Maybe I'm wrong, let's see what Renato says.

Reply:
Yes it's the textarea. It's a fallback if `textManipulation` not explicitly passed.

> Maybe we should just pass some owner to autocomplete where we instanciate it?

In this direction we could just `throw` it here if `textManipulation` is missing, but it's a breaking change - we'll also need to update random `autocomplete()` calls elsewhere, which _may_ be extensively used.

Reply:
Fine to have a fallback, especially since we eventually want to drop this autocomplete implementation.

But... does this fallback actually work? I'm 99% sure DOM elements won't return anything if you call `getOwner()` on them?

Reply:
We weren't even using the other methods that would reach for services, so I've extracted the autocomplete-specific methods to its own class, which doesn't require an owner. @davidtaylorhq @jjaffeux please feel free to re-review. thanks!